## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 444087`
Commit: `9c21d6fd7305f03a7272b822dec3c95edef7eba4`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.tuxedo-drivers</li>
    <li>linuxKernel.packages.linux_5_15.tuxedo-drivers</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_6_1.tuxedo-drivers</li>
    <li>linuxPackages.tuxedo-drivers (linuxKernel.packages.linux_6_12.tuxedo-drivers)</li>
    <li>linuxKernel.packages.linux_hardened.tuxedo-drivers (linuxKernel.packages.linux_6_12_hardened.tuxedo-drivers)</li>
    <li>linuxPackages_latest.tuxedo-drivers (linuxKernel.packages.linux_6_16.tuxedo-drivers)</li>
    <li>linuxKernel.packages.linux_6_6.tuxedo-drivers</li>
    <li>linuxPackages_latest-libre.tuxedo-drivers (linuxKernel.packages.linux_latest_libre.tuxedo-drivers)</li>
    <li>linuxPackages-libre.tuxedo-drivers (linuxKernel.packages.linux_libre.tuxedo-drivers)</li>
    <li>linuxPackages_lqx.tuxedo-drivers (linuxKernel.packages.linux_lqx.tuxedo-drivers)</li>
    <li>linuxPackages_xanmod.tuxedo-drivers (linuxKernel.packages.linux_xanmod.tuxedo-drivers)</li>
    <li>linuxPackages_xanmod_latest.tuxedo-drivers (linuxKernel.packages.linux_xanmod_latest.tuxedo-drivers, linuxPackages_xanmod_stable.tuxedo-drivers)</li>
    <li>linuxPackages_zen.tuxedo-drivers (linuxKernel.packages.linux_zen.tuxedo-drivers)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_10.tuxedo-drivers</summary>
<pre>/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_probe&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:331:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  331 |         const struct platform_device_info pinfo = {
      |         ^~~~~
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_remove&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:360:9: error: implicit declaration of function &#x27;timer_delete&#x27;; did you mean &#x27;timer_reduce&#x27;? [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
  360 |         timer_delete(&amp;profile_changed_timer);
      |         ^~~~~~~~~~~~
      |         timer_reduce
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:361:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  361 |         struct driver_data_t *driver_data = dev_get_drvdata(&amp;wdev-&gt;dev);
      |         ^~~~~~
cc1: some warnings being treated as errors
make[5]: *** [/nix/store/vw58pmiw77ib86s680z2v40s2p4h0mxm-linux-5.10.244-dev/lib/modules/5.10.244/source/scripts/Makefile.build:286: /build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.o] Error 1
make[4]: *** [/nix/store/vw58pmiw77ib86s680z2v40s2p4h0mxm-linux-5.10.244-dev/lib/modules/5.10.244/source/scripts/Makefile.build:503: /build/source/src/tuxedo_nb05] Error 2
make[3]: *** [/nix/store/vw58pmiw77ib86s680z2v40s2p4h0mxm-linux-5.10.244-dev/lib/modules/5.10.244/source/scripts/Makefile.build:503: /build/source/src] Error 2
make[2]: *** [/nix/store/vw58pmiw77ib86s680z2v40s2p4h0mxm-linux-5.10.244-dev/lib/modules/5.10.244/source/Makefile:1851: /build/source] Error 2
make[1]: *** [/nix/store/vw58pmiw77ib86s680z2v40s2p4h0mxm-linux-5.10.244-dev/lib/modules/5.10.244/source/Makefile:192: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/vw58pmiw77ib86s680z2v40s2p4h0mxm-linux-5.10.244-dev/lib/modules/5.10.244/build&#x27;
make: *** [Makefile:29: all] Error 2</pre>
</details>
<details>
<summary>linuxKernel.packages.linux_5_15.tuxedo-drivers</summary>
<pre>      |         ^~~
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_probe&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:331:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  331 |         const struct platform_device_info pinfo = {
      |         ^~~~~
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c: In function &#x27;tuxedo_nb05_power_profiles_remove&#x27;:
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:360:9: error: implicit declaration of function &#x27;timer_delete&#x27;; did you mean &#x27;timer_reduce&#x27;? [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
  360 |         timer_delete(&amp;profile_changed_timer);
      |         ^~~~~~~~~~~~
      |         timer_reduce
/build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.c:361:9: warning: ISO C90 forbids mixed declarations and code [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeclaration-after-statement-Wdeclaration-after-statement8;;]
  361 |         struct driver_data_t *driver_data = dev_get_drvdata(&amp;wdev-&gt;dev);
      |         ^~~~~~
cc1: some warnings being treated as errors
make[4]: *** [/nix/store/65cn8hgqm5ml1hgx783vnsyz7lsgwdak-linux-5.15.193-dev/lib/modules/5.15.193/source/scripts/Makefile.build:289: /build/source/src/tuxedo_nb05/tuxedo_nb05_power_profiles.o] Error 1
make[3]: *** [/nix/store/65cn8hgqm5ml1hgx783vnsyz7lsgwdak-linux-5.15.193-dev/lib/modules/5.15.193/source/scripts/Makefile.build:552: /build/source/src/tuxedo_nb05] Error 2
make[2]: *** [/nix/store/65cn8hgqm5ml1hgx783vnsyz7lsgwdak-linux-5.15.193-dev/lib/modules/5.15.193/source/scripts/Makefile.build:552: /build/source/src] Error 2
make[1]: *** [/nix/store/65cn8hgqm5ml1hgx783vnsyz7lsgwdak-linux-5.15.193-dev/lib/modules/5.15.193/source/Makefile:1926: /build/source] Error 2
make[1]: Leaving directory &#x27;/nix/store/65cn8hgqm5ml1hgx783vnsyz7lsgwdak-linux-5.15.193-dev/lib/modules/5.15.193/build&#x27;
make: *** [Makefile:29: all] Error 2</pre>
</details>
</details>

--- a/nixos/modules/hardware/tuxedo-drivers.nix
+++ b/nixos/modules/hardware/tuxedo-drivers.nix
@@ -31,5 +31,6 @@ in
   config = lib.mkIf cfg.enable {
     boot.kernelModules = [ "tuxedo_keyboard" ];
     boot.extraModulePackages = [ tuxedo-drivers ];
+    services.udev.packages = [ tuxedo-drivers ];
   };
 }

--- a/pkgs/os-specific/linux/tuxedo-drivers/default.nix
+++ b/pkgs/os-specific/linux/tuxedo-drivers/default.nix
@@ -8,6 +8,7 @@
   pahole,
   gitUpdater,
   udevCheckHook,
+  bash,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -26,6 +27,9 @@ stdenv.mkDerivation (finalAttrs: {
 
   postInstall = ''
     echo "Running postInstallhook"
+    substituteInPlace usr/lib/udev/rules.d/* \
+      --replace-quiet "/bin/bash" "${lib.getExe bash}" \
+      --replace-quiet "/bin/sh" "${lib.getExe bash}"
     install -Dm 0644 -t $out/etc/udev/rules.d usr/lib/udev/rules.d/*
   '';
 

